syntax = "proto3";

package iroha.protocol;

import "schema/transaction.proto";
import "schema/response.proto";
import "google/protobuf/empty.proto";


enum TxStatus {
    STATELESS_VALIDATION_FAILED = 0;
    STATELESS_VALIDATION_SUCCESS = 1;
    STATEFUL_VALIDATION_FAILED = 2;
    STATEFUL_VALIDATION_SUCCESS = 3;
    COMMITTED = 4;
    ON_PROCESS = 5;
    NOT_RECEIVED = 6;
}

message ToriiResponse {
    TxStatus tx_status = 1;
}

message TxStatusRequest{
    bytes tx_hash = 1;
}

message FlushRequest {
    string message = 1;
}
message FlushResponse {
    enum FlushStatus {
        SUCCESS = 0;
        FAILED = 1;
    }
    FlushStatus status = 1;
}

service CommandService {
    rpc Torii (Transaction) returns (google.protobuf.Empty);
    rpc Status (TxStatusRequest) returns (ToriiResponse);
    rpc Flush (FlushRequest) returns (FlushResponse);
}


service QueryService {
    rpc Find (Query) returns (QueryResponse);
}